### PR TITLE
fix(rpc): accept jwt's starting with `0x`

### LINKED ISSUE #1586: Support `0x` prefix in JWT secret
### Describe the feature

Currently we don't support a `0x` prefix in the secret, and return the following error if it has a `0x` prefix:
```
+ echo -n 0x7365637265747365637265747365637265747365637265747365637265747365 > /jwt.secret
+ /usr/local/bin/reth node -vvv --db /reth-hive-db --chain /genesis.json --http --http.addr=0.0.0.0 --http.api=admin,debug,eth,net --authrpc.addr=0.0.0.0 --authrpc.jwtsecret=/jwt.secret --nat none
2023-02-28T16:34:33.606422Z  INFO reth 0.1.0 starting
2023-02-28T16:34:33.606741Z  INFO Configuration loaded path=/reth-hive-db
2023-02-28T16:34:33.606750Z  INFO Opening database path=/reth-hive-db
2023-02-28T16:34:33.607319Z  INFO Database opened
2023-02-28T16:34:33.607503Z  WARN warn_msg="No tip specified. reth cannot communicate with consensus clients, so a tip must manually be provided for the online stages with --debug.tip <HASH>."
2023-02-28T16:34:33.607516Z  INFO Consensus engine initialized
2023-02-28T16:34:33.607519Z  INFO Connecting to P2P network
2023-02-28T16:34:33.620790Z  INFO Connected to P2P network peer_id=0xbc66…293f local_addr=0.0.0.0:30303
2023-02-28T16:34:33.620807Z  INFO Test transaction pool initialized
2023-02-28T16:34:33.621014Z  INFO Started RPC server
2023-02-28T16:34:33.621030Z  INFO Engine API handler initialized
Error: Custom error: JWT key is expected to have a length of 64 digits. 66 digits key provided

Location:
    bin/reth/src/node/mod.rs:170:28
```

It would be a nice improvement to allow this prefix, given other clients also accept this format.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by palango**
Resolves #1586

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
please mention this behavior to the docs

lgtm otherwise.

**Comment by palango** [comment]
> please mention this behavior to the docs

Just the inline comments or is there some other place to mention this?

**Comment by mattsse** [comment]
simply add a note to the function doc that this strips leading 0x

File: crates/rpc/rpc/src/layers/jwt_secret.rs [hunk 1]
```diff
@@ -59,7 +59,7 @@ impl JwtSecret {
     /// - `hex` is not a valid hexadecimal string
     /// - `hex` argument length is less than `JWT_SECRET_LEN`
```
Comment (onbjerg) [inline crates/rpc/rpc/src/layers/jwt_secret.rs:60]:
```suggestion
    /// - `hex` argument length is less than `JWT_SECRET_LEN`
    ///
    /// This strips the leading `0x`, if any.
```

**Review by onbjerg — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

**Pull request merged**